src/App.js

```diff
@@ -5,18 +5,32 @@ import "./styles.css";
 
 export default function App() {
   const [dataType, setDataType] = useState("");
-
   const [data, setData] = useState(null);
+  const [loading, setLoading] = useState(false);
 
   console.log({ data });
 
-  // Write code here.
-  //
+  async function getData() {
+    if (!dataType) return;
+    setLoading(true);
+    const response = await fetch("https://swapi.dev/api/" + dataType + "/");
+    const json = await response.json();
+    setData(json);
+    setLoading(false);
```

Comment:
I like what you're doing here, experimenting. Be careful when updating multiple pieces of state, it doesn't always happen in the order you tell it to. In this case, the `loading` state might sometimes get set to false before the `data` state has been updated. If you want to be absolutely sure that one piece of state is updated before another, you can pass a callback function into your `setState` functions like so:

```js
setLoading(() => {
    setData(json)
    return false
})
```

By doing that, you're making sure `setLoading` runs first which will always run `setData` inside of it, then when finished it will return the value you wanted to set your loading state to (in this case, false)